### ISSUE TITLE
Publish CentOS Packages

### ISSUE THREAD

**Issue by apkar** [FoundationDB/fdb-document-layer#6]
[Downloads](https://www.foundationdb.org/download/) page has macOS and Ubuntu packages, but not CentOS. Test and publish CentOS packages.

**Comment by apkar** [comment]
Delay is mostly to do with glibc issues with centos. CentOS has very old `glibc` compared to Ubuntu. It is quite easy to release a package that works for most of the recent Ubuntu versions. Similar issues with `libstdc++`. Of course, this is not an unique problem with this project. We are just trying to figure out the best way to move forward. There are more details at #22. You are welcome to provide any ideas or PRs.

Having said that, I see the value of having an immediate/intermediate solution for the sake of having a packages out there for the community, who doesn’t want to build the project themselves.

**Comment by electricmomo** [comment]
Thanks for your reply.

Actually I have been trying to do a build on Centos 7 with gcc 4.8.5, but with some weird errors that I am tryingt to sort out. So that makes me wish there is a rpm for this.

**Comment by apkar** [comment]
If you are trying to build for CentOS 7, you can use the Docker image. Dockerfile is [here](https://github.com/FoundationDB/fdb-document-layer/blob/master/docker/Dockerfile). This image has all the correct dependencies. Once you build the image, you can use a command like below to build

```
docker run -it -v ~/src/fdb-document-layer:/code -w /code <dockerimage> /bin/bash -c "mkdir build-centos7 && cd build-centos7 && cmake .. && make"
```

Document Layer build depends on FoundationDB, it clones and builds FoundationDB as well. Build might take a while, give it time.

Once the build is done, you will have packages at `build-centos7/packages/rpms/` and binary at `build-centos7/bin`

**Comment by apkar** [comment]
@electricmomo we have published the docker image we use for the build now. It should be easy to build RPM now. More details [here](https://github.com/FoundationDB/fdb-document-layer#build-with-docker).

**Comment by electricmomo** [comment]
Thanks. that worked for me!

**Comment by apkar** [comment]
Centos 7 package available at FoundationDB [download page](https://www.foundationdb.org/download/). Closing the issue, please create another issue if you find any further issues.

cc @electricmomo
